### PR TITLE
Model classes

### PR DESCRIPTION
**Pull request by lensonp**
@Katy-katy 

This requires some changes before merging. Please see new TODO tags and respond to them.

Most importantly, I would like to save the full cross-validation statistics (train/test splits, training objective, objective values for each train/test split, and the mean/std thereof), in a dict, as an attribute of the XRSDModel object that was cross-validated.

Also, should we be using cross-validation during the model.fit()? Right now, we are only using it to select hyperparameters and report model accuracy.

### CODE REVIEW

**Comment by lensonp** [comment]
Also, I did not run tests before submitting this PR- if it fails, I will address the errors tomorrow.

**Comment by Katy-katy** [comment]
@lensonp About "using cross-validation during the model.fit()."   We are using cross-validation to find the best parameters and then we train the model on all available data. Also, .fit() function from sklearn does not have any parameter related with cross-validation: fit(X, y, coef_init=None, intercept_init=None, sample_weight=None)

**Comment by lensonp** [comment]
Ok, I see. 

On "using the same CV as hyperparameters_search()"- what I mean is that maybe the Classifier/Regressor.cross_validate_by_experiments() should use:

cv=model_selection.LeavePGroupsOut(n_groups=n_leave_out).split(
                transformed_data, np.ravel(data_labels), groups=group_by)

... as it is done in hyperparameters_search(). Right now, it looks like cross_validate_by_experiments is set to always leave-one-out, by hand.

**Comment by Katy-katy** [comment]
the saving of downsampling stat is not implemented yet

**Comment by Katy-katy** [comment]
@lensonp review it please
